odinApp/src/odinDetector.cpp

```diff
@@ -226,18 +239,49 @@ int OdinDetector::createOdinDataParams()
 asynStatus OdinDetector::getStatus()
 {
   int status = 0;
+  int acquiring = 0;
 
   // Fetch status items
-  status |= mProcesses->fetch();
-  status |= mFilePath->fetch();
-  status |= mFileName->fetch();
+  status = mParams.fetchAll();
+
+  if (!mAPI.connected()){
+    setIntegerParam(ADStatus, ADStatusDisconnected);
+    setStringParam(ADStatusMessage, "Unable to connect to Odin Server");
+  } else {
+    int status = 0;
+    getIntegerParam(ADStatus, &status);
+    if (status == ADStatusDisconnected){
+      setIntegerParam(ADStatus, ADStatusIdle);
+    }
 
-  status |= mProcessConnected->fetch();
-  status |= mProcessRank->fetch();
-  status |= mWriting->fetch();
-  status |= mFullFileName->fetch();
-  status |= mNumCaptured->fetch();
-  status |= mNumExpected->fetch();
+    // Check if we have initiated an acquisition
+    getIntegerParam(ADAcquire, &acquiring);
+    if (acquiring){
+      // Now fetch the current acquisition status
+      bool acq_state = false;
+      mAcqComplete->get(acq_state);
+      // We are in an acquisition, check the status
+      if (acq_state){
+        // The acquisition has completed, reset the status
+        setIntegerParam(ADAcquire, 0);
+        setStringParam(ADStatusMessage, "Acquisition has completed");
+        setIntegerParam(ADStatus, ADStatusIdle);
+      } else {
+        setStringParam(ADStatusMessage, "Acquiring...");
+      }
+      callParamCallbacks();
+    }
+
+    // Check for any errors
+    // If we are connected to the server then all errors are generated by
+    // the server itself.
+    std::string error_msg = "";
+    mErrorMessage->get(error_msg);
```

Comment:
The IOC crashes here when I enable status polling. It looks like mErrorMessage is not created.